### ISSUE TITLE
Consider capacity for controller_node selection

### ISSUE THREAD

**Issue by AlanCoding** [ansible/awx#10694]
edited by cmeyers:

Today, we randomly choose an online controller node https://github.com/ansible/awx/blob/b5fac4157df2f8c949361a7ca26452f47fc3baf5/awx/main/models/ha.py#L155

We should do "something better" (see comments below for ideas)

**Comment by kdelee** [comment]
We need to:
  - consider the instance capacity when selecting the control node
  - deduct from the instance capacity when we select node as controller
  - release capacity/return capacity to the instance when job is done (when it is done with event processing??)

**Comment by AlanCoding** [comment]
> deduct from the instance capacity when we select node as controller

We talked about deducting 1 unit. That should be sufficient for our purposes. We'd need more data on nodes with control-only nodes to further refine this.

> release capacity/return capacity to the instance when job is done (when it is done with event processing??)

This is pretty much automatic. We compute available capacity for the entire cluster based on the state of jobs using the instances. So when the job goes from pending/running to failed/successful, then it will be removed from consideration.

**Comment by jladdjr** [comment]
We're thinking this is more of a "nice-to-have" item, and will probably get pushed back a bit.

**Comment by kdelee** [comment]
update on current state of PR:

The capacity "impact" of being a controller node is default 1, but is configurable via a setting
In order to be selected to be a control node for a system job/project update, that means we need a minimum of 2 capacity. If a cluster has less than 2 capacity project updates will never run. But for a node to have less than 2 capacity it would need to have such minuscule amount of CPU and RAM...that it practically has 0.
If the control nodes in a cluster have less that 2 available capacity, any jobs in pending will stay in pending. A log to the job_lifecycle logger should indicate the job is waiting on capacity.

The "jobs_running" attribute on instances and instance groups does NOT count controlling jobs as "running jobs" -- only jobs that use node as the execution node/instance group.

The "consumed_capacity" attribute on instances DOES account for controlling jobs

**Comment by kdelee** [comment]
https://github.com/ansible/awx/pull/11665 merged, some docs updates were added to https://github.com/ansible/awx/blob/devel/docs/capacity.md#impact-of-job-types-in-awx and https://github.com/ansible/awx/blob/devel/docs/capacity.md#selecting-the-right-settingsawx_control_node_task_impact

**Comment by AlanCoding** [comment]
If tests are passing in a day or two then this should be a pretty easy close.

**Comment by thenets** [comment]
So far, looks stable. Closing it.
